client/layout/masterbar/logged-in.jsx

```diff
@@ -155,12 +159,20 @@ class MasterbarLoggedIn extends React.Component {
 	}
 
 	render() {
-		const { domainOnlySite, translate, isCheckout, isMigrationInProgress } = this.props;
+		const { domainOnlySite, translate, isCheckout, isMigrationInProgress, siteSlug } = this.props;
 
 		if ( isCheckout === true ) {
 			return (
 				<Masterbar>
 					<div className="masterbar__secure-checkout">
+						<Item
```

Comment:
The only thing that worries me is that `siteSlug` is not a required prop, and if you have a single site and visit `/plans/` with the trailing slash, you're redirected to `/plans//example.com`, which is a routing error and therefore displays a blank page. See #42628

Maybe we could change this to just `/plans` instead if there is no `siteSlug` and let the routing handle the rest?

Reply:
Great catch!